src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -252,5 +257,8 @@ private int ComparePreselection(CompletionItem item1, CompletionItem item2)
 
             return 0;
         }
+
+        internal static string GetDisplayTextForMatching(CompletionItem item)
+            => item.Properties.TryGetValue(DisplayTextForMatching, out var displayText) ? displayText : item.DisplayText;
```

Comment:
i don't quite get how this works.  Who puts the items in the property bag to be read out here?

Reply:
In other words, all i see are reads of this data, no writes anywhere in this PR.  How does htis work?

Reply:
@CyrusNajmabadi Sorry for the lack of context. This is just a workaround for a third-party CompletionProvider that is trying to do some fancy things with the display text. We'll eventually develop better, public APIs to accomplish our shared goals in a more robust way, but for now we're taking a couple tactical fixes to make these experiences better.